lib/puppet-syntax/hiera.rb

```diff
@@ -74,7 +74,7 @@ def check(filelist)
 
       filelist.each do |hiera_file|
         begin
-          yamldata = YAML.load_file(hiera_file)
+          yamldata = YAML.load_file(hiera_file, aliases: true)
```

Comment:
Looks like you need to do some version detection if the `aliases` keyword is supported.